core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala

```diff
@@ -440,6 +472,32 @@ class RequestQuotaTest extends BaseRequestTest {
     assertTrue(s"Throttle time metrics not updated: $client" , throttleTimeMetricValue(clientId) > 0)
   }
 
+  private def checkSmallQuotaProducerkRequestThrottleTime(apiKey: ApiKeys) {
+
+    // Request until throttled using client-id with default small producer quota
+    val smallQuotaProducerClient = Client(smallQuotaProducerClientId, apiKey)
+    val throttled = smallQuotaProducerClient.runUntil(response => responseThrottleTime(apiKey, response) > 0)
+
+    assertTrue(s"Response not throttled: $smallQuotaProducerClient", throttled)
+    assertTrue(s"Throttle time metrics for produce quota not updated: $smallQuotaProducerClient",
+      throttleTimeMetricValueForQuotaType(smallQuotaProducerClientId, QuotaType.Produce) > 0)
+    assertTrue(s"Throttle time metrics for requoest quota updated: $smallQuotaProducerClient",
```

Comment:
Typo requoest. Also, should the throttle time for request rate quota to be larger than 0, since the request rate quota is 0.01?

Reply:
Fixed. The test is instrumented so that both quotas are violated, but throttle time is recorded for the max of the two only. That's why throttle time metrics for request quota is supposed to be 0.